.github/workflows/triage.yml

```diff
@@ -20,42 +19,105 @@ jobs:
-                  project(number: $project_number) {
-                    columns(first: 10) {
+              query($organization: String!, $project_number: Int!) {
+                organization(login: $organization){
+                  projectV2(number: $project_number) {
+                    id
+                    fields(first:20) {
                       nodes {
-                        id,
-                        name
+                        ... on ProjectV2Field {
+                          id
+                          name
+                        }
+                        ... on ProjectV2SingleSelectField {
+                          id
+                          name
+                          options {
+                            id
+                            name
+                          }
+                        }
                       }
                     }
                   }
                 }
               }' > project_columns.json
-          echo 'COLUMN_ID='$(jq '.data.repository.project.columns.nodes[] | select(.name == "Triage") | .id' project_columns.json) >> $GITHUB_ENV
-          echo 'COLUMN_NAME='$(jq '.data.repository.project.columns.nodes[] | select(.name == "Triage") | .name' project_columns.json) >> $GITHUB_ENV
+          echo 'PROJECT_ID='$(
+            jq \
+              '.data.organization.projectV2.id' \
+              project_columns.json
+          ) >> $GITHUB_ENV
+          echo 'STATUS_FIELD_ID='$(
+            jq \
+              '.data.organization.projectV2.fields.nodes[]
+              | select(.name== "Status")
+              | .id' \
+              project_columns.json
+          ) >> $GITHUB_ENV
+          echo 'TRIAGE_COLUMN_ID='$(
+            jq --raw-output \
+              '.data.organization.projectV2.fields.nodes[]
+              | select(.name== "Status")
+              | .options[]
+              | select(.name=="To Triage")
+              | .id' \
+              project_columns.json
+          ) >> $GITHUB_ENV
 
-      - name: Add issue#${{ github.event.issue.number }} to Triage column
+      - name: Add issue#${{ github.event.issue.number }} to project
         env:
           ISSUE_ID: ${{ github.event.issue.node_id }}
         run: |
-          echo "Adding issue: ${{ github.event.issue.number }} to column $COLUMN_NAME in project $PROJECT_NUMBER"
-          if [[ $DRY_RUN == "false" ]]; then
+          echo "Adding issue: ${{ github.event.issue.number }} to project $PROJECT_NUMBER"
+          if [[ $DRY_RUN == 'false' ]]; then
+            item_id="$(
+              gh api graphql \
+                --jq '.data.addProjectV2ItemById.item.id' \
+                --raw-field project_id=$PROJECT_ID \
+                --raw-field issue_id=$ISSUE_ID \
+                --raw-field query='
+                mutation($project_id: ID!, $issue_id: ID!) {
+                  addProjectV2ItemById(input: {projectId: $project_id, contentId: $issue_id}) {
```

Comment:
Does it handle the case where the item is already on the board correctly? What if it's already there but in a different column?

Reply:
Yes. Note that the action does not run for `reopened` issues. Also, if you manually add the issue to another column during its creation, the action will just move it back to the triage column.